views/application.py

```diff
@@ -93,3 +97,37 @@ def update(self, request, *args, **kwargs):
                 list(set(new_team_member_ids)-set(old_team_member_ids))
             )
         return response.Response(serializer.data)
+
+    
+class ApplicationHiddenDetailView(generics.GenericAPIView):
+    """
+    View details of Application which should not be exposed directly 
+    """
+
+    permission_classes = [permissions.IsAuthenticated, ]
+
+    def post(self,request,*args,**kwargs):
+
+        """
+        Authenticate if the requesting user is Team Member
+        of the application and check the password entered.
+        :param request: the request being processed
+        :param args: arguments
+        :param kwargs: keyword arguments
+        :return:response containing client_secret
+        """
+
+        try:
+            application = Application.objects.get(pk = request.data.get('id'))
+            application_team_members = application.team_members.all()
+            for team_member in application_team_members:
+                if team_member.id == request.person.id:
+                    if request.user.check_password(request.data.get('password')):
+                        return response.Response(ApplicationHiddenDetailSerializer(application).data)
+                    else:
+                        return response.Response("Wrong Password",status=403)
+            
+            return response.Response("Requested user is not a team-member",status=403)
+        
+        except:
+            return response.Response("Error",status=404)
```

Comment:
Be more descriptive about the error messages, better to use complete sentences instead of phrases.